staging/src/k8s.io/component-base/controllers/config.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+type Config interface {
+	GetComponentName() string
+	SetComponentName(component string)
+
+	ControllerConfig(controllerName string) (shouldRun bool, withMigration bool)
```

Comment:
The cloud provider would implement this assuming the controller name parameter is either "kube-controller-manager" or "cloud-controller-manager" it will return whether a specific controller should run inside the current component with/without migration lock

Reply:
It maybe helpful to add another abstraction on top of this interface (maybe a builder?) so we can add pile on multiple implementations of ControllerConfig instead of only from the cloud provider. Happy to discuss that part further in the KEP, only focusing on how the components decide to run a controller for now